### PR TITLE
add:nginx service and conf

### PR DESCRIPTION
**Pull request by MysterTech**

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -12,14 +12,14 @@ services:
       - ipfs
       - postgres
     extra_hosts:
```
Comment (zjesko) [inline docker/docker-compose.yml]: we can comment this out, don't need this. Its for connecting to a local RPC which we are not doing

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -43,3 +43,20 @@ services:
...
     volumes:
       - ./data/postgres:/var/lib/postgresql/data
+  nginx:
+    image: nginx:alpine
+    ports:
+      - '80:80'
+      - '443:443'
+    volumes:
+      - ./nginx:/etc/nginx/conf.d/
+      - ./certbot/conf:/etc/letsencrypt
```
Comment (zjesko) [inline docker/docker-compose.yml]: Remove all https/ssl and CORS stuff, we are handling it using AWS ACM and ALB

File: docker/nginx/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,66 @@
+upstream graph-node {
+    server graph-node:8000;
+    server graph-node:8001;
+    server graph-node:8020;
+    server graph-node:8030;
+    server graph-node:8040;
+}
+
+map $http_origin $allow_origin {
```
Comment (zjesko) [inline docker/nginx/nginx.conf]: Remove all https/ssl and CORS stuff from this file too

File: docker/nginx/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,66 @@
+upstream graph-node {
+    server graph-node:8000;
```
Comment (zjesko) [inline docker/nginx/nginx.conf:2]: we don't want to expose all these extra ports, its unsafe

**Review by zjesko — CHANGES_REQUESTED** [review]

**Comment by zjesko** [comment]
You can use this file for reference, (I used it in prod)

```
upstream node {
    server graph-node:8000;
}
upstream health {
    server graph-node:8030;
}

server {

    listen 80;
    server_name api-analytics.canto.neobase.one;

    location / {
        proxy_pass http://node/;
        # proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        # proxy_set_header Host $host;
        # proxy_redirect off;
    }

    location /health/ {
        proxy_pass http://health/;
        # proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        # proxy_set_header Host $host;
        # proxy_redirect off;
    }
}
```

File: docker/nginx/nginx.conf (hunk 4 above)
Comment (MysterTech) [reply docker/nginx/nginx.conf:2]: Originally I had kept just 8000 but after last night's call I changed it to include all. nvm Ill changed it back to the original.

File: docker/docker-compose.yml (hunk 1 above)
Comment (MysterTech) [reply docker/docker-compose.yml]: which specifically?

Comment (zjesko) [reply docker/docker-compose.yml]: `extra_hosts` part

File: docker/nginx/nginx.conf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+map $http_origin $allow_origin {
```
Comment (zjesko) [inline docker/nginx/nginx.conf]: Can you remove this too (CORS stuff)

File: docker/nginx/nginx.conf [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    server_name $hostname;
+
+    client_max_body_size 16M;
+
+    location / {
+        proxy_pass http://graph-node/;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header Host $host;
+        proxy_redirect off;
+	    limit_req zone=applimit burst=20 nodelay;
```
Comment (zjesko) [inline docker/nginx/nginx.conf]: what is the difference between `limit_req` and `limit_req_zone`? are both needed?

File: docker/nginx/nginx.conf [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+map $http_origin $allow_origin {
+    default $http_origin;;
+}
+
+limit_req_zone $binary_remote_addr zone=applimit:10m rate=10r/s;
+limit_req_status 429;
+
+server {
+    listen 8000;
```
Comment (zjesko) [inline docker/nginx/nginx.conf]: why are we listening on 8000? and then mapping the port 80 and 443 in docker compose?

This should be 80 right?

File: docker/nginx/nginx.conf [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+    location / {
+        proxy_pass http://graph-node/;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header Host $host;
+        proxy_redirect off;
+	    limit_req zone=applimit burst=20 nodelay;
+    }
+}
+
```
Comment (zjesko) [inline docker/nginx/nginx.conf:32]: can you add the /health route as well (look at the comment with prod Nginx config)

**Review by zjesko — CHANGES_REQUESTED** [review]

File: docker/nginx/nginx.conf (hunk 6 above)
Comment (jessepinkman9900) [reply docker/nginx/nginx.conf]: https://www.nginx.com/blog/rate-limiting-nginx/

File: docker/nginx/nginx.conf [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+server {
+    listen 80;
+    server_name $hostname;
+
+    location / {
+        proxy_pass http://graph-node/;
+        # proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        # proxy_set_header Host $host;
+        # proxy_redirect off;
+	    limit_req zone=applimit burst=100 nodelay;
```
Comment (zjesko) [inline docker/nginx/nginx.conf:21]: nit: fix indentation. Everything else looks good. Did you test this rate limiting by any chance?

File: docker/nginx/nginx.conf (hunk 8 above)
Comment (MysterTech) [reply docker/nginx/nginx.conf:32]: done.

File: docker/nginx/nginx.conf (hunk 6 above)
Comment (MysterTech) [reply docker/nginx/nginx.conf]: thanks sai. We discussed it on call and I explained what was in this blog to akshat as well.

**Review by zjesko — APPROVED** [review]

**Pull request merged**